src/main/java/mods/railcraft/common/fluids/FluidTools.java

```diff
@@ -107,12 +129,12 @@ private static ProcessState tryFill(IInventory inv, StandardTank tank, ItemStack
 
     private static ProcessState tryDrain(IInventory inv, StandardTank tank, ItemStack container) {
         ItemStack drained = FluidUtil.tryEmptyContainer(container, tank, Fluid.BUCKET_VOLUME, null, true);
-        if (InvTools.isEmpty(drained)) {
+        if (isFailed(drained)) {
             sendToOutput(inv);
             return ProcessState.RESET;
         }
         inv.setInventorySlotContents(1, InvTools.makeSafe(drained));
-        return ProcessState.FILLING;
+        return ProcessState.DRAINING;
```

Comment:
This has bugged me a lot, but I managed to fix this :smile: and one-time containers work then